### ISSUE TITLE
RFC: IPC API v2

### ISSUE THREAD

**Issue by CanadaHonk** [gluon-framework/gluon#29]
Second (major) iteration of IPC API. I dislike having to use an event-based system, I think there should be a wrapper for most developers to use which allows seemingly exposing functions into the web context, either via a function or using a setter in the IPC API object.

#### Implementation A
```js
Window.ipc.expose('key', (arg1, arg2) => {});
```

#### Implementation B
```js
Window.ipc.expose({
  key: (arg1, arg2) => {}
});
```

### Setter
```js
Window.ipc.key = (arg1, arg2) => {};
```

### Example usage
```js
// node
Window.ipc.myFunction = () => {}; // or

Window.ipc.expose('myFunction', () => {}); // or

Window.ipc.expose({
  myFunction: () => {}
});

// web
Gluon.ipc.myFunction();
```

### Function
```js
Window.ipc.unexpose('key');
```

### Setter
```js
delete.Window.ipc.key:
```

### Example usage
```js
delete Window.ipc.key; // or

Window.ipc.unexpose('key');
```

I'm leaning towards using a setter, but it might seem strange/wrong to some people (for the first time). Please comment with which you prefer and/or opinions on all!

**Comment by yellowsink** [comment]
I like the simplicity of a setter, but I fear the "magic" to do it is a bad idea.

Comfiest api though...

**Comment by yellowsink** [comment]
I think i prefer impl A to impl B though of those two.

**Comment by CanadaHonk** [comment]
I suppose we could support both (or all)? Would probably overcomplicate things though.

**Comment by CanadaHonk** [comment]
Adding all 3 implementations described in the issue.

**Comment by WorriedArrow** [comment]
I prefer Implementation B since it allows you to easily define multiple functions in 1 call, so you wouldn't have to keep calling the function.

**Comment by CanadaHonk** [comment]
Thanks for your input. Both implementations are now supported in current branch, as well as the setter.

**Comment by WorriedArrow** [comment]
I like the idea of having two implementations as well. Can't wait to see where this API goes!

**Comment by yellowsink** [comment]
given this is fully implemented, is this rfc closable now?

**Comment by CanadaHonk** [comment]
Will be closed upon v0.10 release shortly.
